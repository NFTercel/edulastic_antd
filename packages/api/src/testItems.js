import API from "./utils/API";

const api = new API();
const prefix = "/testitem";
const prefixElasticSearch = "/search/items";

const formatData = data => {
  const item = JSON.parse(JSON.stringify(data));
  delete item._id;
  return item;
};

const getAll = data =>
  api
    .callApi({
      url: prefixElasticSearch,
      method: "post",
      data
    })
    .then(result => {
      const items = result.data.result.hits.hits.map(el => ({ _id: el._id, ...el._source }));
      const count = result.data.result.hits.total;
      return { items, count };
    });

const getById = (id, params = {}) =>
  api
    .callApi({
      url: `${prefix}/${id}`,
      method: "get",
      params
    })
    .then(result => result.data.result);

const updateById = (id, item) => {
  const { updatedAt, createdAt, authors, autoGrade, ...data } = formatData(item);
  return api
    .callApi({
      url: `${prefix}/${id}`,
      method: "put",
      data
    })
    .then(result => result.data.result);
};

const create = data =>
  api
    .callApi({
      url: prefix,
      method: "post",
      data
    })
    .then(result => result.data.result);

const update = ({ id, item }) => {
  const { updatedAt, createdAt, ...data } = formatData(item);
  return api
    .callApi({
      url: `${prefix}/${id}`,
      method: "put",
      data
    })
    .then(result => result.data.result);
};

const evaluation = (id, answers) =>
  api
    .callApi({
      url: `${prefix}/${id}/evaluation`,
      method: "post",
      data: { answers }
    })
    .then(result => result.data.result);

export default {
  getAll,
  getById,
  updateById,
  create,
  update,
  evaluation
};
