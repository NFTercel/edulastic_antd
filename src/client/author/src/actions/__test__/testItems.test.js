import { receiveTestItemsAction, setTestItemsAction } from "../testItems";

describe("test items", () => {
  it("receive test items should return an action", () => {
    expect(receiveTestItemsAction({})).toMatchSnapshot();
  });

  it("set test items should return an action", () => {
    expect(
      setTestItemsAction([
        {
          index: 1
        },
        {
          index: 2
        },
        {
          index: 3
        }
      ])
    ).toMatchSnapshot();
  });
});
