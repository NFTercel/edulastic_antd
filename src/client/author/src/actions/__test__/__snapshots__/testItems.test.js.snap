// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test items receive test items should return an action 1`] = `
Object {
  "payload": Object {},
  "type": "[testItems] receive items request",
}
`;

exports[`test items set test items should return an action 1`] = `
Object {
  "payload": Object {
    "data": Array [
      Object {
        "index": 1,
      },
      Object {
        "index": 2,
      },
      Object {
        "index": 3,
      },
    ],
  },
  "type": "[testItems] set items request",
}
`;
